sim.py

```diff
@@ -6,69 +6,117 @@
 from scipy.spatial.transform import Rotation as R
 from time import time
 from simulation_manager import logger
+from FSW.controllers.controller import Controller
 import os
+import yaml
+from actuators.magnetorquer import Magnetorquer
+from actuators.reaction_wheels import ReactionWheel
+# from build.world.pyphysics.models import MagneticField, SRP
 
 START_TIME = time()
 
-controller_dt = 0.1
-estimator_dt = 1
-
-
 def controller(state_estimate):
-    return np.array([0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0])
+    
+    return np.array([0, 0, 0, 0, 0, 0, 0])
 
 
 def estimator(y):
     return y
 
+
 def sensors(true_state): # TODO : sensors will return a shorter measurement vector and estimator will estimate the full state
-    return true_state + np.array([5000, 5000, 5000, 5, 5, 5, 0.05, 0.05, 0.05, 0.05, 2, 2, 2, 2])*np.random.random()
+    return true_state # + np.array([5000, 5000, 5000, 5, 5, 5, 0.05, 0.05, 0.05, 0.05, 2, 2, 2, 2])*np.random.random()
 
 
 def run(log_directory, config_path):
     logr = logger.MultiFileLogger(log_directory)
 
-    params = SimParams(config_path)
+    params = SimParams(config_path) # cpp params
+    with open(config_path, "r") as f:
+        pyparams = yaml.safe_load(f)
 
     initial_state = np.array(params.initial_state) # Get initial state
+    newstate = np.array(rk4(initial_state, np.zeros((params.num_MTBs + params.num_RWs, 1)), params, -params.dt, params.dt))
```

Comment:
This needs to go into C++ initializeSatellite() function

Reply:
cant make it work. Ill need help on this, Ive spent too much time and its not critical